numba/stencilparfor.py

```diff
@@ -269,6 +269,69 @@ def _mk_stencil_parfor(self, label, in_args, out_arr, stencil_ir,
                                        self.calltypes)
             equiv_set.insert_equiv(out_arr, in_arr_dim_sizes)
             init_block.body.extend(stmts)
+        else: # out is present
+            if "cval" in stencil_func.options: # do out[:] = cval
+                cval = stencil_func.options["cval"]
+                # TODO: Loosen this restriction to adhere to casting rules.
```

Comment:
I thk this can be resolved before merging since you can check the casting with `self.typingctx`.

Reply:
I think that make sense to have `out` dominate.  But, I thought the comment was simply about `return_type.dtype != typing.typeof.typeof(cval)` requiring an exact match where it could have checked if `cval` is castable to `return_type.dtype`